openapi/src/main/java/io/micronaut/openapi/visitor/AbstractOpenApiVisitor.java

```diff
@@ -990,6 +997,39 @@ protected Schema bindSchemaForElement(VisitorContext context, Element element, C
                 if (schemaDescription.isPresent()) {
                     schemaToBind.setDescription(schemaDescription.get());
                 }
+                Optional<String> schemaDefaultValue = schemaAnn.get("defaultValue", String.class);
+                if (schemaDefaultValue.isPresent()) {
+                    try {
+                        schemaToBind.setDefault(jsonMapper.readValue(schemaDefaultValue.get(), Map.class));
+                    } catch (JsonProcessingException e) {
+                        schemaToBind.setDefault(schemaDefaultValue.get());
+                    }
+                }
+                Optional<String> schemaExample = schemaAnn.get("example", String.class);
+                if (schemaExample.isPresent()) {
+                    try {
+                        schemaToBind.setExample(jsonMapper.readValue(schemaExample.get(), Map.class));
+                    } catch (JsonProcessingException e) {
+                        schemaToBind.setExample(schemaExample.get());
+                    }
+                }
+                Optional<Boolean> schemaDeprecated = schemaAnn.get("deprecated", Boolean.class);
+                if (schemaDeprecated.isPresent()) {
+                    schemaToBind.setDeprecated(schemaDeprecated.get());
+                }
+                Optional<io.swagger.v3.oas.annotations.media.Schema.AccessMode> schemaAccessMode = schemaAnn.get("accessMode", io.swagger.v3.oas.annotations.media.Schema.AccessMode.class);
+                if (schemaAccessMode.isPresent()) {
+                    if (schemaAccessMode.get() == io.swagger.v3.oas.annotations.media.Schema.AccessMode.READ_ONLY) {
+                        schemaToBind.setReadOnly(true);
+                        schemaToBind.setWriteOnly(null);
+                    } else if (schemaAccessMode.get() == io.swagger.v3.oas.annotations.media.Schema.AccessMode.WRITE_ONLY) {
+                        schemaToBind.setReadOnly(false);
+                        schemaToBind.setWriteOnly(null);
+                    } else if (schemaAccessMode.get() == io.swagger.v3.oas.annotations.media.Schema.AccessMode.READ_WRITE) {
+                        schemaToBind.setReadOnly(null);
+                        schemaToBind.setWriteOnly(null);
+                    }
+                }
```

Comment:
Not critical but this code can be written more cleanly with something like:

```java
schemaAnn.enumValue("accessMode", io.swagger.v3.oas.annotations.media.Schema.AccessMode.class)
   .ifPresent(accessMode -> {
          switch(accessMode) {
               case READ_ONLY:
                        schemaToBind.setReadOnly(true);
                         schemaToBind.setWriteOnly(null);
                break;
               case WRITE_ONLY:
                        schemaToBind.setReadOnly(false);
                         schemaToBind.setWriteOnly(null);
                break;
          }
   });
```

Apart from that is setting write only to null correct?

Reply:
@graemerocher I don't now, soory. This nlock copypasted from swagger lib

Reply:
@graemerocher Your proposal is not so simple, because would require making the schemaToBind variable finalized. In this case, I think that this block is better left as is.

![изображение](https://user-images.githubusercontent.com/48474279/173856194-485d33ae-22d6-4ea5-b596-9722a58565db.png)